### DISCUSSION TITLE
Get result from Template::render?

### DISCUSSION THREAD

**Discussion by b1rger** [rwf2/Rocket#2044] [Questions]
Is it possible to get a result from `Template::render`? Lets say I want to fallback to another template if Template::render fails somehow or I want to test multiple templates and return the first that renders successfully? Or would I have to use `tera` or `handlebars` directly in that case?

**Comment by edgarogh** [comment]
[`Template::show`](https://api.rocket.rs/v0.5-rc/rocket_dyn_templates/struct.Template.html#method.show) returns an option.

~~The following solution is a bit hacky, but because `Template` [implements `Responder`](https://api.rocket.rs/v0.5-rc/rocket_dyn_templates/struct.Template.html#impl-Responder%3C%27r%2C%20%27static%3E) (and defers error reporting to the responding phase), you could manually call [`respond_to`](https://api.rocket.rs/v0.5-rc/rocket/response/trait.Responder.html#tymethod.respond_to). If it returns an `Err`, you know that the template couldn't be rendered.~~ This doesn't give you any useful error object to print/analyse though.

**Comment by SergioBenitez** [comment]
You could use https://api.rocket.rs/master/rocket_dyn_templates/struct.Metadata.html#method.render, but this is only available on `master` for now.
